lib/src/client/call.dart

```diff
@@ -282,13 +292,18 @@ class ClientCall<Q, R> implements Response {
       // should contain "trailers" as well (Trailers-Only).
       _trailers.complete(_headerMetadata);
       final status = _headerMetadata['grpc-status'];
-      // If status code is missing, we must treat it as '0'. As in 'success'.
       final statusCode = status != null ? int.parse(status) : 0;
+
       if (statusCode != 0) {
         final message = _headerMetadata['grpc-message'] == null
             ? null
             : Uri.decodeFull(_headerMetadata['grpc-message']);
-        _responseError(GrpcError.custom(statusCode, message));
+
+        _responseError(GrpcError.custom(
+          statusCode,
+          message,
+          decodeStatusDetails(_headerMetadata['grpc-status-details-bin']),
+        ));
```

Comment:
I suggest extracting this into a common helper because this chunk of code repeats two times, maybe something along the lines of:

```
void _checkForErrorStatus(Map<String, String> metadata) {
  if (!metadata.containsKey('grpc-status')) {
    return;
  }

  final status = int.parse(metadata['grpc-status']);
  // the rest of the code.
  
}
```

Reply:
Fixed in https://github.com/bottlepay/grpc-dart/commit/cd197160a01b0b1ac641eccd433d5b4f55f20f9c